recipes/libpng/all/conanfile.py

```diff
@@ -131,38 +127,47 @@ def _libpng_cmake_system_processor(self):
             return "powerpc"
         return str(self.settings.arch)
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["PNG_TESTS"] = False
-        self._cmake.definitions["PNG_SHARED"] = self.options.shared
-        self._cmake.definitions["PNG_STATIC"] = not self.options.shared
-        self._cmake.definitions["PNG_DEBUG"] = self.settings.build_type == "Debug"
-        self._cmake.definitions["PNG_PREFIX"] = self.options.api_prefix
-        self._cmake.definitions["CMAKE_MACOSX_BUNDLE"] = False # prevents configure error on shared iOS/tvOS/watchOS
+    def layout(self):
+        cmake_layout(self)
```

Comment:
I advice to put source code in a subdir `src` to avoid mixing-up patches and source code

Reply:
Done. Added src_folder.